spec/mailers/communicart_mailer_spec.rb

```diff
@@ -225,4 +225,18 @@ def sender_names(mail)
       expect(sender_names(mail)).to eq(["Communicart"])
     end
   end
+
+  describe '#proposal_subject' do
+    it 'defaults when no client_data is present' do
+      proposal = FactoryGirl.create(:proposal)
+      mail = CommunicartMailer.proposal_created_confirmation(proposal)
+      expect(mail.subject).to eq("Request " + proposal.public_id)
+    end
+
+    it 'includes custom text for ncr work orders' do
+      wo = FactoryGirl.create(:ncr_work_order)
+      mail = CommunicartMailer.proposal_created_confirmation(wo.proposal)
+      expect(mail.subject).to eq("Request #{wo.public_identifier}, #{wo.org_code}, #{wo.building_number}, from #{wo.requester.email_address}")
```

Comment:
Might want to consider setting explicit values when creating the instance and checking for explicit values here.